WordPress/gutenberg
# Fix: inconsistent field spacing in Grid layout

The inclusion of icons / avatars in field display has caused mis-alignment in Grid layout:

<img width="1497" alt="Screenshot 2024-07-10 at 16 19 17" src="https://github.com/WordPress/gutenberg/assets/846565/b63f1711-4b85-4659-95f4-2ee18a8cbca6">

The first two fields are 24px tall, the third is only 16px tall. The result is uncanny (and too large) spacing between.

This PR adds a `min-height` to the field row, and reduces the gap between:

<img width="1497" alt="Screenshot 2024-07-10 at 16 12 00" src="https://github.com/WordPress/gutenberg/assets/846565/450ef8af-ce26-4e33-88ce-e0cb4b57a38b">



## Testing Instructions
1. Navigate to the Pages data view and switch to Grid layout
2. Toggle the date field on in view options
3. Observe equal spacing between each field
